relayer/src/error.rs

```diff
@@ -554,7 +555,76 @@ impl GrpcStatusSubdetail {
             .trim_start()
             .starts_with("account sequence mismatch")
     }
+
+    pub fn is_account_sequence_mismatch_that_can_be_ignored(&self) -> bool {
+        match parse_sequences_in_mismatch_error_message(self.status.message()) {
+            None => false,
+            Some((expected, got)) => expected < got,
```

Comment:
Is this different from the usual case which the expected sequence is larger than the provided sequence? And is this safe to ignore because the actual SendTX allows provided account sequence to be higher than the next available sequence?

It may worth adding a comment on why the inverse can be ignored.

Reply:
> Is this different from the usual case which the expected sequence is larger than the provided sequence? And is this safe to ignore because the actual SendTX allows provided account sequence to be higher than the next available sequence?

Yes and yes. The details are in the issue.
expected > got = typically happens if some other agent uses the same wallet as the relayer, will happen in simulate and broadcast; we have to refresh in this case
expected > go = typically happens around the time when tm switches to a new block and we try to simulate a Tx after the new block and app state are created but while mempool Tx-es are "rechecked".  When this happens we can ignore the simulation failure and broadcast the Tx. The checkTx for this will happen after all other mempool Tx-es are rechecked so the mismatch error should not be seen anymore.

> It may worth adding a comment on why the inverse can be ignored.
tried in 132cd46276230574d3d0299b6192615cfe01cc78